File: 프로그래머스/lv1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(left, right):
    answer = 0
    
    for num in range(left, right+1):
        cnt = 0                     # 약수의 갯수
        for i in range(1, num+1):
            if num % i == 0:        # 약수에 해당하면 카운트
                cnt += 1
        if cnt % 2:                 # 약수가 홀수개라면 빼고 
            answer -= num
        else:                       # 약수가 짝수개라면 더함
            answer += num
            
    return answer

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.java
class Solution {
    public int solution(int left, int right) {
        int answer = 0;
        
        for (int i = left; i <= right; i++) {
            int cnt = 1;        // 1은 반드시 자기 자신의 약수
            
            for (int j = 2; j <= i; j++) {
                if (i % j == 0) {
                    cnt++;
                }
            }
            
            if (cnt % 2 == 0) {
                answer += i;
            } else {
                answer -= i;
            }
            
        }
        return answer;
    }
}